book/src/rfcs/0001-rfc-process.md

```diff
@@ -1 +1,105 @@
-# RFC-0001: RFC Process
+# RFC-0001: RFC process
+
+* Status: Draft
+* Pull request: https://github.com/asterinas/asterinas/pull/2365/
+* Date submitted: 2025-08-24
+* Date approved: YYYY-MM-DD
+
+## Summary
+
+The "RFC" (request for comments) process is intended to provide a consistent, transparent, structured path for the community to make "big" decisions. For example, the RFC process can be used to evolve the project roadmap and the system architecture.
+
+## Motivation
+
+As the Asterinas project grows in scale and community size, an informal decision-making process becomes insufficient. A formalized process is crucial for maintaining the quality and coherence of the project. Several highly successful open-source projects have demonstrated the value of such a process, including Rust ([RFCs](https://rust-lang.github.io/rfcs/introduction.html)), Fuchsia ([RFCs](https://fuchsia.dev/fuchsia-src/contribute/governance/rfcs)), Python ([Python Enhancement Proposals](https://peps.python.org/) or PEPs), and Kubernetes ([Kubernetes Enhancement Proposals](https://www.kubernetes.dev/resources/keps/) or KEPs).
+
+Adopting a formal RFC process will:
+- **Enhance decision quality**: By requiring a detailed proposal and public discussion, we can ensure that decisions are well-reasoned, technically sound, and consider a wide range of perspectives.
+- **Foster Consensus**: The process provides a clear framework for the community to come together, discuss differing viewpoints, and arrive at a consensus on important issues.
+- **Create design records**: Approved RFCs will serve as a historical archive, documenting the design choices made and the reasoning behind them. This is invaluable for new contributors and for future reference.
+
+By introducing this process, we aim to support the healthy and sustainable growth of the Asterinas project.
+
+## Design
+
+### When to follow the RFC process
+
+The RFC process should be used for any "substantial" change to Asterinas.
+
+Examples of changes that **require** an RFC include:
+- Launching new top-level sub-projects (e.g., OSTD, OSDK).
+- Defining or significantly altering the project roadmap and goals.
+- Establishing or changing project-wide norms, such as this RFC process itself or coding style guides.
+- Proposing significant architectural designs, such as the framekernel architecture or safe policy injection.
+- Introducing changes that affect user-space programs, such as adding a non-Linux system call.
```

Comment:
I know this list isn't supposed to be exhaustive, but there's one important point I think is worth mentioning:
* Defending a particular solution when there are non-trivial trade-offs involved

Reply:
> Defending a particular solution when there are non-trivial trade-offs involved.

Of course. Such a solution should be justified by the author, but it may not be done in the form of an RFC. If the solution solved a "local" or "unimportant" problem, then the solution itself wouldn't meet the bar of a "significant" change.